File: src/main/java/com/jackmeng/stl/stl_Populable.java
// Software created by Jack Meng (AKA exoad). Licensed by the included "LICENSE" file. If this file is not found, the project is fully copyrighted.

package com.jackmeng.stl;

public abstract interface stl_Populable< T >
{
  void add_1(T e);

  T add_2(T e);

  void add_3(T e, int i);

  T add_4(T e, int i);

  void remove_1(T e);

  int remove_2(T e);

  void remove_3(int i);

  int remove_4(int i);

  default int len()
  {
    return 0;
  }

  default void remove_all()
  {
    int x = len();
    while(x -- > 0) remove_3(x);
  }
}

File: src/main/java/com/jackmeng/stl/stl_Populable.kt
// Software created by Jack Meng (AKA exoad). Licensed by the included "LICENSE" file. If this file is not found, the project is fully copyrighted.
package com.jackmeng.stl

interface stl_Populable<T>
{
	fun add_1(e:T)
	fun add_2(e:T):T
	fun add_3(e:T , i:Int)
	fun add_4(e:T , i:Int):T
	fun remove_1(e:T)
	fun remove_2(e:T):Int
	fun remove_3(i:Int)
	fun remove_4(i:Int):Int
	fun len():Int
	{
		return 0
	}
	
	fun remove_all()
	{
		var x=len()
		while (x-->0) remove_3(x)
	}
}